tensorflow/lite/micro/compression/model_editor.py

```diff
@@ -0,0 +1,572 @@
+
+  def __init__(self,
+               shape,
+               dtype,
+               buffer=None,
+               data=None,
+               quantization=None,
+               name=None):
+    """Initialize Tensor.
+
+        Args:
+            shape: Tensor shape as tuple
+            dtype: TensorType enum value
+            buffer: Optional Buffer object (for explicit buffer sharing)
+            data: Optional numpy array or bytes (convenience parameter, creates Buffer)
+            quantization: Optional Quantization object
+            name: Optional tensor name
+
+        Raises:
+            ValueError: If both buffer and data are specified
+        """
+    if data is not None and buffer is not None:
+      raise ValueError("Cannot specify both data and buffer")
+
+    self.shape = shape
+    self.dtype = dtype
+    self.buffer = buffer
+    self.quantization = quantization
+    self.name = name
+    self._index = None
+
+    # Convert data to buffer if provided
+    if data is not None:
+      buf_data = data if isinstance(data, bytes) else data.tobytes()
+      self.buffer = Buffer(data=buf_data)
+
+  @property
+  def array(self) -> Optional[np.ndarray]:
+    """Get tensor data as properly-shaped numpy array.
+
+        Returns:
+            numpy array with shape matching tensor.shape and dtype matching
+            tensor.dtype, or None if tensor has no data.
+
+        For low-level byte access, use tensor.buffer.data instead.
+        """
+    if self.buffer is None:
+      return None
+    return np.frombuffer(self.buffer.data,
+                         dtype=_dtype_to_numpy(self.dtype)).reshape(self.shape)
+
+  @array.setter
+  def array(self, value: np.ndarray):
+    """Set tensor data from numpy array.
+
+        Args:
+            value: New tensor data as numpy array. Will be converted to bytes
+                   using tobytes() and stored in the buffer.
+
+        Creates a new Buffer if tensor has no buffer, or updates the existing
+        buffer's data in place.
+
+        For low-level byte access, use tensor.buffer.data instead.
+        """
+    buf_data = value.tobytes()
+    if self.buffer is None:
+      self.buffer = Buffer(data=buf_data)
+    else:
+      self.buffer.data = buf_data
+
+  @property
+  def index(self) -> Optional[int]:
+    """Tensor index in the subgraph's tensor list.
+
+        Returns index after read() or build(). May be None or stale after
+        modifications. Use with caution.
+        """
+    return self._index
+
+  @property
+  def numpy_dtype(self) -> np.dtype:
+    """Get numpy dtype corresponding to tensor's TFLite dtype.
+
+        Returns:
+            numpy dtype object for use with np.frombuffer, np.array, etc.
+        """
+    return _dtype_to_numpy(self.dtype)
+
+
+@dataclass
+class OperatorCode:
+  """Operator code specification."""
+  builtin_code: tflite.BuiltinOperator
+  custom_code: Optional[str] = None
+  version: int = 1
```

Comment:
must maintain `deprecated_builtin_code`, and it must interoperate as per the schema.fbs comments with `builtin_code`.